### PR TITLE
BLD: time stamp and build number update

### PR DESCRIPTION
**Pull request by ke-zhang-rd**
Update time stamp to 2019C3.0 and code style. Validate there is a space before "0" of version.

### CODE REVIEW

**Comment by mrakitin** [comment]
@ke-zhang-rd, could you please rebase on master and resolve the conflicts before it gets reviewed? Thanks.

**Comment by ke-zhang-rd** [comment]
> Please note, our approach is the same as [conda-forge's one](https://conda-forge.org/docs/maintainer/updating_pkgs.html#bumping-the-build-number) - once the version is updated, the build number should be reset to `0`. Could you please make these changes where necessary?

Two place were updated.
https://github.com/NSLS-II/lightsource2-recipes/pull/704/files#diff-5b703f5cf37d0871171a050180a93002R10

https://github.com/NSLS-II/lightsource2-recipes/pull/704/files#diff-799a608ba25d9f1e8728138901843a64R10

File: recipes-tag/23-id-1-csx1-collection/meta.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {% set year = "2019" %}
-{% set cycle = "2" %}
-{% set version ="0" %}
+{% set cycle = "3" %}
+{% set version = "0" %}
```
Comment (mrakitin) [inline recipes-tag/23-id-1-csx1-collection/meta.yaml]: For this package the dependency of `23-id-1-csx1-analysis >=2019C2.0` should be better enforced, I think, i.e. `23-id-1-csx1-analysis >=2019C3.0`.

File: recipes-tag/pympler/meta.yaml [hunk 2]
```diff
@@ -44,4 +44,4 @@ about:
 
 extra:
   recipe-maintainers:
-    - CJ-Wright
\ No newline at end of file
+    - CJ-Wright
```
Comment (mrakitin) [inline recipes-tag/pympler/meta.yaml]: Why do we need this update?

Comment (ke-zhang-rd) [reply recipes-tag/pympler/meta.yaml]: Check https://thoughtbot.com/blog/no-newline-at-end-of-file
vim add those for me. It isn't what I want, but looks like a good feature.

File: recipes-tag/23-id-1-csx1-collection/meta.yaml (hunk 1 above)
Comment (ke-zhang-rd) [reply recipes-tag/23-id-1-csx1-collection/meta.yaml]: I see.

File: recipes-tag/pympler/meta.yaml (hunk 2 above)
Comment (mrakitin) [reply recipes-tag/pympler/meta.yaml]: I think it's OK to do it if there are other changes in the file, but in this case it looks like an unintended change. Also, adding a new line at the end of a _Python_ file is discouraged by PEP-8. I suggest to revert the changes, as we don't update this recipe's version/build number.

Comment (tacaswell) [reply recipes-tag/pympler/meta.yaml]: Where does pep8 advise against adding a newline to the last line of a file (which is different than an extra black line).  We are comparing `'stuff'`, `'stuff\n'`, and `'stuff\n\n'`.  Before this PR the file was in the first state and this is moving it to the second.

POSIX says we need a `\n` on every line: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206

While it is a bit thrashy to do this as part of this PR, I do not think we should revert it.

Comment (mrakitin) [reply recipes-tag/pympler/meta.yaml]: Here is a simple demonstration of the PEP-8 enforcement in flake8:
```bash
$ cat test-blank-line.py
import os
print(os.__file__)
$ flake8 test-blank-line.py
$ echo '' >> test-blank-line.py
$ cat test-blank-line.py
import os
print(os.__file__)

$ flake8 test-blank-line.py
test-blank-line.py:3:1: W391 blank line at end of file
```

Comment (mrakitin) [reply recipes-tag/pympler/meta.yaml]: OK, I agree with @tacaswell we can keep the change. In vim I don't see any difference on my mac. In the hex-mode I see the following differences:
### in master:
```bash
$ hexdump meta.yaml | tail -10
0000400 5f 75 72 6c 3a 20 68 74 74 70 3a 2f 2f 70 79 74
0000410 68 6f 6e 68 6f 73 74 65 64 2e 6f 72 67 2f 50 79
0000420 6d 70 6c 65 72 2f 0a 20 20 64 65 76 5f 75 72 6c
0000430 3a 20 68 74 74 70 73 3a 2f 2f 67 69 74 68 75 62
0000440 2e 63 6f 6d 2f 70 79 6d 70 6c 65 72 2f 70 79 6d
0000450 70 6c 65 72 0a 0a 65 78 74 72 61 3a 0a 20 20 72
0000460 65 63 69 70 65 2d 6d 61 69 6e 74 61 69 6e 65 72
0000470 73 3a 0a 20 20 20 20 2d 20 43 4a 2d 57 72 69 67
0000480 68 74
0000482
```
### in this PR branch:
```bash
$ hexdump meta.yaml | tail -10
0000400 5f 75 72 6c 3a 20 68 74 74 70 3a 2f 2f 70 79 74
0000410 68 6f 6e 68 6f 73 74 65 64 2e 6f 72 67 2f 50 79
0000420 6d 70 6c 65 72 2f 0a 20 20 64 65 76 5f 75 72 6c
0000430 3a 20 68 74 74 70 73 3a 2f 2f 67 69 74 68 75 62
0000440 2e 63 6f 6d 2f 70 79 6d 70 6c 65 72 2f 70 79 6d
0000450 70 6c 65 72 0a 0a 65 78 74 72 61 3a 0a 20 20 72
0000460 65 63 69 70 65 2d 6d 61 69 6e 74 61 69 6e 65 72
0000470 73 3a 0a 20 20 20 20 2d 20 43 4a 2d 57 72 69 67
0000480 68 74 0a
0000483
```
so the [`0a`](https://www.cisco.com/c/en/us/td/docs/ios/12_4/cfg_fund/command/reference/cfnapph.html) (LF, line feed) symbol is added. The advantage is that the file is displayed fine when `cat meta.yml` is used, i.e.:
```bash
✔ ~/src/lightsource2-recipes/recipes-tag/pympler [master|✔]
17:15 $ cat meta.yaml | tail -3
extra:
  recipe-maintainers:
    - CJ-Wright✔ ~/src/lightsource2-recipes/recipes-tag/pympler [master|✔]
17:15 $ git checkout time-stamp
Switched to branch 'time-stamp'
Your branch is up-to-date with 'ke-zhang-rd/time-stamp'.
✔ ~/src/lightsource2-recipes/recipes-tag/pympler [time-stamp|✔]
17:15 $ cat meta.yaml | tail -3
extra:
  recipe-maintainers:
    - CJ-Wright
✔ ~/src/lightsource2-recipes/recipes-tag/pympler [time-stamp|✔]
17:15 $
```

**Comment by ke-zhang-rd** [comment]
> Thanks @ke-zhang-rd! It looks better now, but I don't fully understand why we only update the version of the meta-packages, but not the dependencies? It's make sense to also update the dependencies (if any) in one shot, so we don't have to re-iterate with bumping the build number if there are new dependencies.

If we don't update the version of dependency package, we may leave the old build number. If we update the version, build number will reset to 0.

**Comment by mrakitin** [comment]
@ke-zhang-rd, my point was to also add the updated dependencies here, if there are any updates. I see no value of bumping version/build number of the metapackages without updating their dependencies.

**Pull request merged**